hub/core/tensor.py

```diff
@@ -622,3 +627,35 @@ def tobytes(self) -> bytes:
 
     def _pop(self):
         self.chunk_engine._pop()
+
+    def _append_to_links(self, sample, flat: Optional[bool]):
+        for k, v in self.meta.links.items():
+            if flat is None or v["flatten_sequence"] == flat:
+                self.dataset[k].append(get_link_transform(v["append"])(sample))
+
+    def _update_links(
+        self,
+        global_sample_index: int,
+        sub_index: Index,
+        new_sample,
+        flat: Optional[bool],
+    ):
+        for k, v in self.meta.links.items():
+            if flat is None or v["flatten_sequence"] == flat:
```

Comment:
why do we need both of these? if flatten sequence was already specified at the time of linking, we don't need flat arg anymore right?

Reply:
It is still required.. some links can be flat, others not.